scoringutilhelpers/R/join_forecast_and_data.R

```diff
@@ -0,0 +1,28 @@
+library(arrow)
+library(readr)
+library(dplyr)
+
+#' Join Forecast and Actual Data
+#'
+#' This function reads forecast data from a Parquet file and actual data from a
+#' TSV file, then joins them using a specified key.
+#'
+#' @param forecast_source A character vector specifying the path to the
+#' directory containing Parquet file(s) containing forecast data.
+#' @param data_path A character string specifying the path to the TSV file
+#' containing actual/truth data.
```

Comment:
`forecast_source` mentions "character vector", while `data_path` mentions "character string"

Reply:
Also a bit uncertain about the "directory containing Parquet file(s)". I understand this is a feature of `arrow`, but I'm not sure we have committed to setting things up in a way that is compatible with this expectation.

Reply:
I'm pretty sure that by using parquet this is just a feature with no downsides?

Reply:
I haven't used this feature so you could be right. I was wondering what happens if you give it a directory with multiple parquet files that are not part of the same dataset?

Reply:
You seem to be able to select with the options, part of my motivation to have the `arrow` function the target of splatting here.